cmd/dynamic_startup.go

```diff
@@ -213,6 +213,24 @@ func DynamicStartPreInit(nodeConfig *NodeConfig) error {
 
 	// set the root snapshot in the config - we will use this later to bootstrap
 	nodeConfig.RootSnapshot = snapshot
+
+	// set the FullBlockHeight to be rootHeight+TransactionExpiry.
```

Comment:
This approach works. I'd be happy to merge this as-is to the relevant `v0.xx` branch to quickly unblock the current AN issues. For merging to master though, I think it would be better to adjust the responsibility for handling this edge case. Adding some suggestions below 👇 

In the current implementation in this PR, the DynamicStartup function implements mitigating this edge case by initializing the `last_full_block_height` index. I think this is non-ideal because:
* The `DynamicStartup` feature is not at all related to the `ingestion` engine which manages the `last_full_block_height` index
* The `DynamicStartup` feature is used by all node roles (not just ANs, which are the only role where this mitigation is relevant)

I think it would be better for the `ingestion` engine to internally handle the additional initialization logic for the `last_full_block_height` index. `ingestion.Engine` implements `Component`, so we can implement this in the engine's `Start` method:
```go 
func (e *Engine) Start(parent irrecoverable.SignalerContext) {
	rootBlock := e.state.Params().Root()
	rootSnapshot := e.state.AtBlockID(rootBlock.ID())
	if !protocol.IsSporkRootSnapshot(rootSnapshot) {
		// this is useful for dynamically bootstrapped access node, they will request missing collections. In order to ensure all txs
		// from the missing collections can be verified, we must ensure they are referening to known blocks.
		// That's why we set the full block height to be rootHeight + TransactionExpiry, so that we only request missing collections
		// in blocks above that height.
		firstFullHeight := rootBlock.Height + flow.DefaultTransactionExpiry
		e.blocks.UpdateLastFullBlockHeight(firstFullHeight)
	}
	
	e.Start(parent)	
}